src/gui/application.cpp

```diff
@@ -619,10 +621,21 @@ void Application::parseOptions(const QStringList &options)
             _versionOnly = true;
         } else if (option == QLatin1String("--language")) {
             if (it.hasNext() && !it.peekNext().startsWith(QLatin1String("--"))) {
-                _userEnforcedLanguage = it.next();
+                auto languageParam = it.next();
+
+                // fail if the language is unknown
+                if (!Translations::listAvailableTranslations().contains(languageParam)) {
+                    std::cerr << "Error: unknown language " << languageParam.toStdString() << std::endl;
```

Comment:
Hm as the errors here do not got though displayHelpText they might not be discoverable on Windows.
Since we nowadays attach to the console its not that big of an issue but...

Reply:
I can also use `displayHint` as before, which will `std::exit(...)`. However, then, I have to duplicate listing the available languages.

Why not reference `--list-languages` from this message? Then we can use `displayHint` just fine.

Reply:
go for it

Reply:
or you could define a function/lambda that you call :D

Reply:
Let's conclude that #8603 will make everything a lot better...